versionutils/version.go

```diff
@@ -84,8 +85,13 @@ func (v *Version) IsGreaterThan(lesser *Version) bool {
 		return false
 	}
 
-	if lesser.ReleaseCandidate > 0 {
-		if v.ReleaseCandidate == 0 || v.ReleaseCandidate > lesser.ReleaseCandidate {
+	if lesser.LabelVersion > 0 {
+		if v.LabelVersion == 0 {
+			return true
+		} else if lesser.Label != v.Label {
+			// 1.0.0-fooX is not greater than or less than 1.0.0-barY
+			return false
```

Comment:
false implies that it is less than or equal to, but as the comment indicates, this version is neither greater nor less than the other one

this is a confusing/misleading api as written. an error might be clearer

Reply:
just returning false drops the information that the comparison was impossible.

the user should be forced to decide how to handle incomparable versions and handle the appropriate exported error